tour_model_eval/get_scores.py

```diff
@@ -15,8 +15,20 @@ def compare_trip_orders(bins,bin_trips,filter_trips):
 
 
 # This function is to get homogeneity score after the first/second round of clustering
-# It is based on bin_trips, which are common trips. bin_trips are collected according to the indices in bins
+# It is based on bin_trips, which are common trips. bin_trips are collected according to the indices of the trips
+# in bins above cutoff
 # More info about bin_trips is in similarity.py (delete_bins)
+# For mapping the labels, in one cluster, we do have trips with user labels.
+# There are only two situations - trips are grouped in one cluster or the single trip in one cluster.
+# Ideally, we should have labels [11,11,11] for trips in one cluster after the second round.
+# In such a cluster, we should already have 1 or 2 trips that have user labels, the third one is predicted.
+# Then we label it with the same user labels as the previous two.
+# We don't have 1.0 homogeneity score for most of the cluster. It can also be [11,11,12].
+# Then for the third one, we will still label it with the same user labels as the previous two
+# (since they are in one cluster).
```

Comment:
I think you need to fix this example. Even with `[11,11,12]`, the homogeneity score will be 1.0 because

> Non-perfect labelings that further split classes into more clusters can be perfectly homogeneous:

Also, "we don't have 1.0 homogeneity score" is an artifact of the data, not of the algorithm.

Reply:
In fact, this makes it seem that we will always have a 1.0 homogeneity score because we will have `[11,11,11]` or [11,11,12]`

Reply:
@corinne-hcr Added suggested text below; please fill in and commit.